keylime/requests_client.py

```diff
@@ -75,7 +66,32 @@ class HostNameIgnoreAdapter(HTTPAdapter):
     It is required because in most cases we don't know the hostname during certificate generation.
     """
 
+    _tls_context = None
+    _ignore_hostname = False
+
+    def __init__(self, tls_context, ignore_hostname, *args, **kwargs):
+        self._tls_context = tls_context
+        self._ignore_hostname = ignore_hostname
+
+        super().__init__(*args, **kwargs)
+
     def init_poolmanager(self, connections, maxsize, block=requests.adapters.DEFAULT_POOLBLOCK, **pool_kwargs):
-        self.poolmanager = PoolManager(
-            num_pools=connections, maxsize=maxsize, block=block, strict=True, assert_hostname=False, **pool_kwargs
-        )
+        if self._ignore_hostname and self._tls_context:
+            self.poolmanager = PoolManager(
+                num_pools=connections,
+                maxsize=maxsize,
+                block=block,
+                strict=True,
+                assert_hostname=False,
```

Comment:
you can remove the `if` and the other branch with `assert_hostname=not (self._ignore_hostname and self._tls_context)` or creating a variable (maybe more clear) like `assert_hostname = False if self._ignore_hostname and self._tls_context else True`.

Reply:
Thanks for the suggestion! The only nit is that `assert_hostname` does not accept `True`, but only `False`, `None`, or the actual hostname.